task7/script.js

```diff
@@ -0,0 +1,369 @@
+        }
+
+        addAll(posts) {
+            let notValidPosts;
+            posts.forEach((post) => {
+                if (ChirikList.validatePost(post)) {
+                    ChirikList.updateHashtags(post);
+                    this._posts.push(post);
+                } else
+                    notValidPosts.push(post);
+            });
+            console.log("Added " + this._posts.size + " elements, " + notValidPosts.size + " elements not valid");
+
+            return notValidPosts;
+        }
+
+        getPosts(skip, top, filterConfig) {
+            skip = skip || 0;
+            top = top || 10;
+
+            let result = this._posts.slice();
+
+            if (filterConfig.hashtags) {
+                result = result.filter(function (item) {
+                    if (item.hashtags === null || item.hashtags === undefined)
+                        return false;
+                    for (let i = 0; i < filterConfig.hashtags.length; i++) {
+                        if (item.hashtags.includes(filterConfig.hashtags[i]))
+                            return true;
+                    }
+                });
+            }
+
+            if (filterConfig.timeSort) {
+                result = result.sort((a, b) => a.createdAt - b.createdAt);
+            } else if (filterConfig.nameSort) {
+                result = result.sort((a, b) => a.name > b.name);
+            }
+
+            console.log("get _posts, skipping " + skip + ", getting " + top + ", filtered by: "
+                + "name:" + filterConfig.nameSort + ", time:" + filterConfig.timeSort +
+                ", with hashtags: " + (filterConfig.hashtags || "none") + "\n");
+            return result.slice(skip, skip + top);
+        }
+
+        getPost(id) {
+            let result = this._posts.find(element => element.id === id);
+            console.log("Find post with id:" + id + ", found: " + result + "\n");
+            return result;
+        }
+
+        static validatePost(post) {
+            console.log(post + " : ")
+            if (post.id === undefined) {
+                console.log("wrong id\n");
+                return false;
+            }
+            if (post.description === undefined || post.description.length >= 200) {
+                console.log("wrong description\n");
+                return false;
+            }
+            if (post.createdAt === undefined || !(post.createdAt instanceof Date)) {
+                console.log("wrong time\n");
+                return false;
+            }
+            if (post.author === undefined) {
+                console.log("wrong author\n");
+                return false;
+            }
+            if (post.name === undefined) {
+                console.log("wrong name\n");
+                return false;
+            }
+            return true;
+        }
+
+        static updateHashtags(post) {
+            let hashtagRegExp = /#\S+/;
+            post.hashtags = hashtagRegExp.exec(post.description);
```

Comment:
Incorrect, exec find fist matching and returns array of regexp group
e.g. /#(\s+)/.exec('Hi #hash tag #example')  // ['#hash', 'hash'] - where [full match, 1 group]

Reply:
Done.